### PR TITLE
[TorchInductor] Fix batch_aten_sub dropping alpha argument during post-grad fusion

### LINKED ISSUE #195442: [TorchInductor] `batch_aten_sub` miscompilation drops `alpha`
### 🐛 Describe the bug

[TorchInductor] `batch_aten_sub` miscompilation drops `alpha`

The `batch_aten_sub` post-grad fusion incorrectly removes the `alpha=2` argument when batching multiple `aten.sub.Tensor` operations.

## Complete reproducer

```python
import torch
from torch._inductor import config
from torch._dynamo.utils import counters


class Model(torch.nn.Module):
    def forward(self, x0, y0, x1, y1, x2, y2, x3, y3, x4, y4):
        z0 = torch.ops.aten.sub.Tensor(x0, y0, alpha=2)
        z1 = torch.ops.aten.sub.Tensor(x1, y1, alpha=2)
        z2 = torch.ops.aten.sub.Tensor(x2, y2, alpha=2)
        z3 = torch.ops.aten.sub.Tensor(x3, y3, alpha=2)
        z4 = torch.ops.aten.sub.Tensor(x4, y4, alpha=2)
        return z0, z1, z2, z3, z4


device = "cuda"
inputs = []
for i in range(5):
    x = torch.full((16, 16), 10.0 + i, device=device)
    y = torch.full((16, 16), 3.0 + i, device=device)
    inputs.extend([x, y])

model = Model().eval()
eager_out = model(*inputs)

torch._dynamo.reset()
counters.clear()

with config.patch({
    "post_grad_fusion_options": {"batch_aten_sub": {}}
}):
    compiled_model = torch.compile(model, backend="inductor", fullgraph=True)
    compiled_out = compiled_model(*inputs)

print("batch_aten_sub count:", counters["inductor"]["batch_aten_sub"])
for ref, out in zip(eager_out, compiled_out):
    print(ref[0, 0].item(), out[0, 0].item())

torch.testing.assert_close(compiled_out, eager_out)
```

## Expected result

Each operation computes `x - 2 * y`. Therefore the expected first element of the five outputs is:

```text
[4.0, 3.0, 2.0, 1.0, 0.0]
```

## Actual result

The fusion is triggered and combines the five operations into one batched subtraction. However, the post-grad graph becomes:

```python
sub_tensor = torch.ops.aten.sub.Tensor(cat, cat_1)
```

The `alpha=2` argument is missing. The generated kernel performs `cat - cat_1` instead of `cat - 2 * cat_1`. Thus the actual first element of all five compiled outputs is:

```text
[7.0, 7.0, 7.0, 7.0, 7.0]
```

The compiled result is therefore different from eager execution, and `torch.testing.assert_close` fails.

### Versions

pytorch version: 2.15.0 e289091

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by PranshulSoni**
Fixes #195442

### Description
In `BatchPointwiseMathOpsPostGradFusion.fuse()` (`torch/_inductor/fx_passes/group_batch_fusion.py`), `kwargs` was hardcoded to `{"alpha": alpha} if self.op == aten.add.Tensor else {}`. When `batch_aten_sub` post-grad horizontal fusion matched `aten.sub.Tensor` operations with a non-default `alpha` (e.g. `alpha=2.0`), `kwargs` defaulted to `{}` in the fused call. Consequently, the batched subtraction computed `x - y` instead of `x - alpha * y`, leading to silent numerical divergence.

This PR:
1. Passes `subset[0].kwargs` directly to `graph.call_function` in `BatchPointwiseMathOpsPostGradFusion.fuse()`.
2. Forwards `**kwargs` when evaluating `batch_op.meta["val"] = self.op(stack_inputs_meta, stack_others_meta, **kwargs)` so the fused node's metadata is computed with the same arguments it will be executed with (hygiene; `alpha` does not affect shape/dtype/device meta).
3. Keys the fusion group on `str(node.kwargs)` in `match()` (mirroring `BatchMathOpsPreGradFusion`), which guarantees the kwargs copied by `fuse()` are uniform across the subset.
4. Adds `test_pointwise_math_ops_with_alpha_fusion` and `test_batch_aten_sub_preserves_alpha` to `test/inductor/test_group_batch_fusion.py` to verify forward correctness, input gradient equivalence, and multi-group `alpha` separation.

### Test Plan
Ran local regression test:
```bash
python -m unittest test.inductor.test_group_batch_fusion.TestGroupBatchFusion.test_batch_aten_sub_preserves_alpha
```
Ran linter and formatter:
```bash
ruff check torch/_inductor/fx_passes/group_batch_fusion.py test/inductor/test_group_batch_fusion.py
ruff format --check torch/_inductor/fx_passes/group_batch_fusion.py test/inductor/test_group_batch_fusion.py
```
Verified that `batch_aten_sub` with non-default `alpha` produces exact numerical equality with eager execution.

> **AI Disclosure:** An AI coding assistant was used to assist in locating the regression and drafting the unit test. The root cause, architectural fix, and test verification were independently validated.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by PranshulSoni** [comment]
Addressed all items in 43edfa92bba:

1. (blocking lint) The PYFMT violation is gone - the imports moved to module scope, and I verified both files with ruff format --check and ruff check locally.

2. (group key) match() now keys on str(node.kwargs), mirroring BatchMathOpsPreGradFusion, and the alpha/rounding_mode locals are dropped - rounding_mode was dead code since batch_aten_div registers aten.div.Tensor. fuse() carries a comment stating the invariant (key must encode everything fuse() copies).

3. (commit message) Rewritten; the meta["val"] forwarding is now described as hygiene, not a divergence source.

4./5. (test cleanup) The test is now test_batch_aten_sub_preserves_alpha with a loop-based forward, module-scope imports, no recompile(), and no dead counters.clear() calls.

7. (absent vs explicit alpha=1) I traced this case and the premise does not hold: the dispatcher elides kwargs equal to their schema defaults, so sub(x, y) and sub(x, y, alpha=1) both trace to kwargs={} - there is no fusion opportunity to lose. I added test_batch_aten_sub_normalizes_default_alpha to pin that normalization behavior instead.

Testing: ran the two CPU tests against a nightly-torch venv with the patch applied (both pass), reverted the venv module to the pre-fix code (test_batch_aten_sub_preserves_alpha fails with {2.0} vs {2.0, 3.5} as expected), and ran the full TestGroupBatchFusion class (20 tests, OK; the 15 GPU-only ones are skipped locally and covered by CI).

**Comment by PranshulSoni** [comment]
@claude review again

**Review by sylvesterkaczmarek — COMMENTED** [review]
Keying the fusion group on the full kwargs before propagating `subset[0].kwargs` closes the actual correctness hole. Using the same kwargs for the fake/meta call matters too; otherwise graph metadata can disagree with runtime semantics. The two-alpha case pins the important grouping boundary.

**Comment by PranshulSoni** [comment]
Thanks for looking, the kwargs grouping check makes sense.

@sylvesterkaczmarek @jansel torchtitan_models_integration looks unrelated, it is just qwen3_5_moe_fsdp+tp+ep: No module named 'fla' and DrCI flags it as broken trunk on merge base. All inductor suites green on my side. Happy to retry that job if you want, otherwise could you approve if this looks good?

**Comment by jansel** [comment]
@claude review again

**Review by jansel — APPROVED** [review]
